File: csharp/Pub/Pub.cs
namespace Pub
{
    public class Pub
    {
        public const string OneBeer = "hansa";
        public const string OneCider = "grans";
        public const string AProperCider = "strongbow";
        public const string Gt = "gt";
        public const string BacardiSpecial = "bacardi_special";

        public int ComputeCost(string drink, bool student, int amount)
        {

            if (amount > 2 && (drink == Gt || drink == BacardiSpecial))
            {
                throw new Exception("Too many drinks, max 2.");
            }
            int price;
            if (drink == OneBeer)
            {
                price = 74;
            }
            else if (drink == OneCider)
            {
                price = 103;
            }
            else if (drink == AProperCider) price = 110;
            else if (drink == Gt)
            {
                price = Ingredient6 + Ingredient5 + Ingredient4;
            }
            else if (drink == BacardiSpecial)
            {
                price = Ingredient6 / 2 + Ingredient1 + Ingredient2 + Ingredient3;
            }
            else
            {
                throw new Exception("No such drink exists");
            }
            if (student && (drink == OneCider || drink == OneBeer || drink == AProperCider))
            {
                price = price - price / 10;
            }
            return price * amount;
        }

        //one unit of rum
        private int Ingredient1 { get; } = 65;

        //one unit of grenadine
        private int Ingredient2 { get; } = 10;

        //one unit of lime juice
        private int Ingredient3 { get; } = 10;

        //one unit of green stuff
        private int Ingredient4 { get; } = 10;

        //one unit of tonic water
        private int Ingredient5 { get; } = 20;

        //one unit of gin
        private int Ingredient6 { get; } = 85;
    }
}

File: python/pub/pub.py
ONE_BEER = "hansa"
ONE_CIDER = "grans"
A_PROPER_CIDER = "strongbow"
GT = "gt"
BACARDI_SPECIAL = "bacardi_special"


def compute_cost(drink: str, is_student: bool, amount: int):
    if amount > 2 and (drink == GT or drink == BACARDI_SPECIAL):
        raise ValueError('Too many drinks, max 2.')
    price = 0
    if drink == ONE_BEER:
        price = 74
    elif drink == ONE_CIDER:
        price = 103
    elif drink == A_PROPER_CIDER:
        price = 110
    elif drink == GT:
        price = ingredient6() + ingredient5() + ingredient4()
    elif drink == BACARDI_SPECIAL:
        price = ingredient6() / 2 + ingredient1() + ingredient2() + ingredient3()
    else:
        raise ValueError('No such drink exists')
    if is_student and (drink == ONE_CIDER or drink == ONE_BEER or drink == A_PROPER_CIDER):
        price = price - price / 10
    return price * amount


# One unit of rum
def ingredient1():
    return 65


# One unit of grenadine
def ingredient2():
    return 10


# One unit of juice
def ingredient3():
    return 10


# One unit of green stuff
def ingredient4():
    return 10


# One unit of tonic water
def ingredient5():
    return 20


# One unit of gin
def ingredient6():
    return 85
